src/runtime_src/core/edge/drm/zocl/zocl_cu.c

```diff
@@ -73,10 +75,11 @@ void
 zocl_cu_check(struct zocl_cu *cu)
 {
 	struct zcu_tasks_info tasks;
-
-	cu->funcs->check(cu->core, &tasks);
-	cu->done_cnt += tasks.num_tasks_done;
-	cu->ready_cnt += tasks.num_tasks_ready;
+	if(((cu->prop & IP_CONTROL_MASK) >> IP_CONTROL_SHIFT) == AP_CTRL_CHAIN || cu->done_cnt == 0) {
```

Comment:
You could add a protocol field in struct zocl_cu and define control protocol in zocl_cu.h.
When scheduler initial a zcu, it could read from xclbin and set zcu object.

The basic rule is don't coupling cu and xclbin.